### ISSUE TITLE
Calculating enthalpy and entropy at exactly the middle value between min and max temperature does not work

### ISSUE THREAD

**Issue by fwitte** [CoolProp/CoolProp#2295]
Hi all,

I have discovered a as I find rather specific/weird bug (?). I need half the value as starting point for some newton iterations in case I do not have a better guess on where to start.

Calculating enthalpy or entropy after updating the state of incompressibles at a temperature which is exactly the middle value between minimum and maximum temperature does not work for many fluids. You can use the snippet below to reproduce.

```python

import CoolProp as CP


p = 1e6
for fluid in ["PHE", "Water", "TVP1", "DowQ"]:
    state = CP.AbstractState("INCOMP", fluid)

    Tmax = state.trivial_keyed_output(CP.iT_max)
    Tmin = state.trivial_keyed_output(CP.iT_min)
    state.update(CP.PT_INPUTS, p, (Tmax + Tmin) *  0.5)
    try:
        state.hmass()
    except ValueError as e:
        msg = (
            "Exactly the middle between maximum and minimum temperature does "
            f"not work for {fluid}."
        )
        print(msg)

        # rhomass or other properties are no problem
        state.rhomass()
        state.T()
        # changing it slightly makes it work
        state.update(CP.PT_INPUTS, p, (Tmax + Tmin) *  0.499999999)
        state.hmass()
        state.update(CP.PT_INPUTS, p, (Tmax + Tmin) *  0.500000001)
        state.hmass()
```
Also, maybe someone can hint me how to call this for every incompressible available in CoolProp? 

In any way, thanks a lot for the awesome work, have a nice day!

CoolProp version: 6.5.0.post1
Python version: 3.11.4
OS: Windows 11

**Comment by ibell** [comment]
I have not the foggiest idea, perhaps you hit exactly the reference state? I seem to recall there was some issue with evaluation AT the reference state.

@jowr?

**Comment by msaitta-mpr** [comment]
Poking around, the issue is trying to evaluate at `T_base`. This happens to be the midpoint of the temperature range (presumably for reasons discussed at http://www.coolprop.org/fluid_properties/Incompressibles.html#thermodynamics-of-incompressible-fluids). 

The crash eventually occurs in `Polynomial2DFrac::evaluate` when the denominator has, effectively, `T - T_base`. I am not sure what the easy solution for this is. The first thing that comes to mind is just checking for `T == T_base` and then evaluating at `T = T+ DBL_EPSILON`.

**Comment by jowr** [comment]
That is all correct - it is actually mentioned in the blue box on the homepage at the end of the first section http://www.coolprop.org/fluid_properties/Incompressibles.html#general-introduction 

Back in the days I had to replace some ordinary polynomials with rational polynomials to allow for a better approximation, but I did not fully comprehend the consequences. Especially entropy turned out to be a major PITA, but we solved it to some extent.

I agree with @msaitta-mpr - the quick fix would be to avoid the division by zero. I would say that one could interpolate instead of just bumping up the temperature.
